lib/system.nim

```diff
@@ -2542,8 +2542,40 @@ template spliceImpl(s, a, L, b: untyped): untyped =
   # fill the hole:
   for i in 0 ..< b.len: s[a+i] = b[i]
 
-template `^^`(s, i: untyped): untyped =
-  (when i is BackwardsIndex: s.len - int(i) else: int(i))
+template `^^`*(s: typed; i: BackwardsIndex): int =
```

Comment:
I find `^^` to be a weird operator for this purpose (I mean, just look at this: `s ^^ ^1`, it doesn't look like something that make a lot of sense). Since this is not the kind of operator that has to be short (only used by `[^x]` and friends), I think it should have a name instead. I don't have any good name ideas now, though.

Reply:
any suggestions? I also prefer names.

Reply:
How about `toIndex(s, i)`? Though that's probably too generic...

Reply:
toIndex is fine, the export in system is the issue, only a few library authors would really need this.